sips/sip-021/sip-021-nakamoto.md

```diff
@@ -0,0 +1,556 @@
+
+This bootup procedure is amenable to activating this proposal from Stacks 2.x, as long as it happens on a reward cycle boundary.
+When the node has synchronized to the latest reward cycle, it would run this algorithm to discover new tenures within the current reward cycle until it reaches the chain tip. Once it has processed all blocks as of the last sortition, it continues to keep pace with the current miner by receiving new blocks broadcasted through the peer network by Stackers once they accept the next Stacks block.
+
+## Mining Protocol
+
+![Figure 2][figure-2-asset]
+
+*Figure 2: The mining run-loop. Mining has two phases; sortition and block creation. To attempt to win a sortition a miner will submit a block-commit and monitor the Bitcoin blockchain to determine if it has won. Once a miner has won the current tenure the miner will proceed to submit new blocks to the Stackers for approval, which will be incorporated into the chain if the block is successfully signed by the Stackers.*
+
+## Block Structure
+### Block Header
+Amongst other things, the original Stacks header is designed to:
+
+1. Tie a single Stacks block to a single Bitcoin block with the sortition that chose the current miner.
+2. Handle appending the next Stacks blocks to a microblock.
+
+The Nakamoto proposal changes the functionality of the Stacks Blockchain to utilize tenure changes as a  consensus critical artifact that links sortitions to the chosen miner as well as removes the support for microblocks. As such, the following is the updated block header to support the features in this proposal and remove support for those being deprecated:
+
+The Nakamoto block header wire format  is as follows.
+|Name|Description|Representation|
+|-|-|-|
+|version|Version number to describe how to validate the block.|1 byte|
+|chain length|The total number of StacksBlock and NakamotoBlocks preceding this block in this block's history.|8 bytes, big-endian|
+|burn spent|Total amount of BTC spent producing the sortition that selected the miner whose tenure produced this block.|8 bytes, big-endian|
+|consensus hash|The consensus hash of the burnchain block that selected this tenure.  The consensus hash uniquely identifies this tenure, including across all Bitcoin forks.|20 bytes|
+|parent block ID|The index block hash of the immediate parent of this block. This is the hash of the parent block's hash and consensus hash.|32 bytes|
+|transaction Merkle root|The SHA512/256 root hash of the binary Merkle tree calculated over the sequence of transactions in this block.|32 bytes|
+|state index Merkle root|The SHA512/256 root hash of the MARF once all of the contained transactions are processed.|32 bytes|
+|miner signature|Recoverable ECDSA signature from the tenure's miner.|65 bytes|
+|stacker signature|A Schnorr signature collectively generated by the set of Stackers over this block|65 bytes|
+
+**Total bytes:** 263
```

Comment:
```suggestion
**Total bytes:** 262
```

Reply:
Updated because of the bitvec addition to 267 + ceil(num_stackers / 8)